posts/google-summer-of-code-fellows/gsoc-2024-fellows.qmd

```diff
@@ -0,0 +1,294 @@
+# GENERAL CONFIGURATION
+###################################
+ENV["JULIA_EDITOR"] = "nvim"
+
+###################################
+# Revise.jl
+###################################
+enable_autocomplete_brackets(false)
+
+###################################
+# HELPER FUNCTIONS
+###################################
+function template()
+    Template(;
+        user="jay-sanjay",
+        dir="~/FOSS",
+        authors="jaysanjay <jaysanjay@gmail.com> and contributors",
+        julia=v"1.6",
+        plugins=[
+            ProjectFile(; version=v"0.0.1"),
+            Git(),
+            Readme(),
+            License(; name="MIT"),
+            GitHubActions(; extra_versions=["1.6", "1", "nightly"]),
+            TagBot(),
+            Codecov(),
+            Documenter{GitHubActions}(),
+            Citation(; readme = true),
+            RegisterAction(),
+            BlueStyleBadge(),
+            Formatter(;style = "blue")
+        ],
+    )
+end
+
+```
+
+
+**4. Selecting treatments of interest**
+
+So, as a part of this, we used the previously mentioned research paper and discussion with the mentors we came up with logic for it. The first thing to do was to determine the moment in time from which selected treatments of interest should be included in the treatment pathway. The default is all treatments starting after the index date of the target cohort. For example, for a target cohort consisting of newly diagnosed patients, treatments after the moment of first diagnosis are included. However, it would also be desirable to include (some) treatments prior to the index date, for instance in case a specific disease diagnosis is only confirmed after initiating treatment. Therefore, periodPriorToIndex specifies the period (i.e. number of days) prior to the index date from which treatments should be included. We have created two dispatches for this function.
+After that proper testing and documentation are also added.
+
+A basic implementation for it is:
+(1) Construct a SQL query to select cohort_definition_id, subject_id, and cohort_start_date from a specified table, filtering by cohort_id.
+(2) Executes the constructed SQL query using a database connection, fetching the results into a data frame.
+(3) If the DataFrame is not empty, convert cohort_start_date to DateTime and subtract date_prior from each date, then return the modified DataFrame.
+
+This was then be called like this:
+```julia
+period_prior_to_index(
+        cohort_id = [1, 1, 1, 1, 1], 
+        conn; 
+        date_prior = Day(100), 
+        tab=cohort
+    )
+```
+
+
+**5. Filters applied**
+
+After this, we where needed to get the patient's database filtered more finely so that there are minimal variations that can be ignored. The duration of the above extracted event eras may vary a lot and it can be preferable to limit to only treatments exceeding a minimum duration. Hence, minEraDuration speciﬁes the minimum time an event era should last to be included in the analysis. 
```

Comment:
Could we  mention that we are using DataFrames.jl here for an initial implementation?